src/core/events/file.rs

```diff
@@ -1,42 +1,94 @@
 //! Handles the file (json) to Rust event retrieval and the unmarshaling process.
 
-use std::{
-    collections::HashMap,
-    fs::File,
-    io::{BufRead, BufReader},
-    path::Path,
-    time::Duration,
-};
-
-use anyhow::{anyhow, Result};
+use std::{collections::HashMap, fs::File, io::BufReader, path::Path, time::Duration};
+
+use anyhow::{anyhow, bail, Result};
 use log::debug;
 
-use super::{Event, EventFactory, EventResult, SectionFactories};
+use super::*;
 use crate::module::ModuleId;
 
 /// File events factory retrieving and unmarshaling events
 /// parts.
-pub(crate) struct FileEventsFactory {
-    reader: BufReader<File>,
+pub(crate) struct FileEventsFactory<'a> {
+    iter: serde_json::StreamDeserializer<
+        'a,
+        serde_json::de::IoRead<BufReader<File>>,
+        serde_json::Value,
+    >,
     factories: SectionFactories,
+    metadata_factories: MetadataFactories,
 }
 
-impl FileEventsFactory {
-    pub(crate) fn new<P>(file: P) -> Result<Self>
+impl FileEventsFactory<'_> {
+    pub(crate) fn new<P>(file: P, metadata_factories: MetadataFactories) -> Result<Self>
```

Comment:
What's the use case for allowing the modules to insert module-specific metadata all scattered throughout the event file? What kind of data cannot be stored in an actual event or in a global (single) metadata section?

Reply:
Also, please comment on #388 as it goes in the opposite direction regarding Factories.